### PR TITLE
re2 - 20190801-

### PR DESCRIPTION
**Pull request by JPeterMugaas**
Update to llatest version
build BOTH static and shared versions of re2

### CODE REVIEW

File: mingw-w64-re2/PKGBUILD [hunk 1]
```diff
@@ -31,30 +31,53 @@ prepare() {
 }
 
 build() {
-  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
-  mkdir -p "${srcdir}/build-${MINGW_CHOST}"
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  msg2 "Test static for static ${MINGW_CHOST}"
```
Comment (Alexpux) [inline mingw-w64-re2/PKGBUILD]: what this is a message?

File: mingw-w64-re2/PKGBUILD [hunk 2]
```diff
@@ -31,30 +31,53 @@ prepare() {
...
     -G "MSYS Makefiles" \
     -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
     -DCMAKE_BUILD_TYPE=Release \
-    -DRE2_BUILD_TESTING=OFF \
+    -DRE2_BUILD_TESTING=ON \
     -DBUILD_SHARED_LIBS=ON \
     ../${_realname}-${_pkgver}
 
   make -j1
+   msg2 "Test static for static ${MINGW_CHOST}"
```
Comment (Alexpux) [inline mingw-w64-re2/PKGBUILD]: The same

File: mingw-w64-re2/PKGBUILD [hunk 3]
```diff
@@ -31,30 +31,53 @@ prepare() {
...
+  msg2 "Test static for static ${MINGW_CHOST}"
+  cd "${srcdir}/build-${MINGW_CHOST}-static"
 
-  #make test
+  make test
 }
 
 package() {
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  cd "${srcdir}/build-${MINGW_CHOST}-shared"
```
Comment (Alexpux) [inline mingw-w64-re2/PKGBUILD]: First you need install static and then shared as for many static and shared packages  there are some difference in headers that will prevent proper shared linking

File: mingw-w64-re2/PKGBUILD (hunk 1 above)
Comment (JPeterMugaas) [reply mingw-w64-re2/PKGBUILD]: Thanks for pointing out my error.  I have taken the liberty of fixing the messages as you pointed out.  I use those messages as progress indicators and to help identify problems with particular build variations.  Ditto for the other one.

File: mingw-w64-re2/PKGBUILD (hunk 3 above)
Comment (JPeterMugaas) [reply mingw-w64-re2/PKGBUILD]: Thanks for pointing that out.  I have made the correction as you suggested.

File: mingw-w64-re2/PKGBUILD [hunk 4]
```diff
@@ -31,30 +31,53 @@ prepare() {
 }
 
 build() {
-  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
-  mkdir -p "${srcdir}/build-${MINGW_CHOST}"
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  msg2 "Build static for ${MINGW_CHOST}"
```
Comment (Alexpux) [inline mingw-w64-re2/PKGBUILD]: This is shared build.

**Comment by Alexpux** [comment]
@JPeterMugaas fix comment

File: mingw-w64-re2/PKGBUILD [hunk 5]
```diff
@@ -31,54 +31,51 @@ prepare() {
 }
 
 build() {
-  msg2 "Build static for ${MINGW_CHOST}"
-  [[ -d "${srcdir}/build-${MINGW_CHOST}-shared" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}-shared"
-  mkdir -p "${srcdir}/build-${MINGW_CHOST}-shared"
-  cd "${srcdir}/build-${MINGW_CHOST}-shared"
-
-  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+  msg2 "Build static for static ${MINGW_CHOST}"
```
Comment (Alexpux) [inline mingw-w64-re2/PKGBUILD]: "static for static" ?

**Comment by Alexpux** [comment]
@JPeterMugaas wil you ever finish this without errors? Look at next comment

**Comment by JPeterMugaas** [comment]
I'm unable to do this without creating problems.  Basically, the way the package works is:

1. build static
2. build shared
3. test static
4. test shared
5. install static
6. install shared

I reordered those to try to give a consistent feel.

You can either accept this as is or I could alternatively close this request and resubmit.

**Comment by Alexpux** [comment]
@JPeterMugaas what you can’t do? Can’t fix msg?

**Comment by JPeterMugaas** [comment]
I can't fix the comit notes (message).  Is that what you meant?

**Comment by Alexpux** [comment]
@JPeterMugaas no I mean fix message in PKGBUILD, you wrote “static for static”

**Pull request merged**